src/internal/_toString.js

```diff
@@ -17,15 +17,15 @@ module.exports = function _toString(x, seen) {
     case '[object Array]':
       return '[' + _map(recur, x).join(', ') + ']';
     case '[object Boolean]':
-      return typeof x === 'object' ? 'new Boolean(' + recur(x.valueOf()) + ')' : x.toString();
+      return typeof x === 'object' ? 'new Boolean(' + _toString(x.valueOf(), []) + ')' : x.toString();
     case '[object Date]':
       return 'new Date(' + _quote(_toISOString(x)) + ')';
     case '[object Null]':
       return 'null';
     case '[object Number]':
-      return typeof x === 'object' ? 'new Number(' + recur(x.valueOf()) + ')' : 1 / x === -Infinity ? '-0' : x.toString(10);
+      return typeof x === 'object' ? 'new Number(' + _toString(x.valueOf(), []) + ')' : 1 / x === -Infinity ? '-0' : x.toString(10);
     case '[object String]':
-      return typeof x === 'object' ? 'new String(' + recur(x.valueOf()) + ')' : _quote(x);
+      return typeof x === 'object' ? 'new String(' + _toString(x.valueOf(), []) + ')' : _quote(x);
     case '[object Undefined]':
       return 'undefined';
     default:
```

Comment:
Thanks for raising this issue, @CrossEye. There is a connection. Here's the definition of `recur`:

```javascript
var recur = function recur(y) {
  var xs = seen.concat([x]);
  return _indexOf(xs, y) >= 0 ? '<Circular>' : _toString(y, xs);
};
```

To evaluate `R.toString(new Number(0))`, we end up evaluating `_indexOf([new Number(0)], 0)` which *should* give `-1` but actually gives `0`. As a result we return `'new Number(<Circular>)'` rather than `'new Number(0)'`. Once #1099 has been merged I'll rebase and revert this workaround.